### PR TITLE
feat: add object notification docs

### PR DESCRIPTION
**Pull request by garrensmith**
This resolves https://linear.app/tigrisdata/issue/TIG-2624/documentation-for-object-notifications

### CODE REVIEW

**Review by rbarabas — APPROVED** [review]

File: docs/objects/object_notifications.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# Object Notifications
```
Comment (ovaistariq) [inline docs/objects/object_notifications.md:1]: I would put this document page under the `buckets` directory as the notifications are applicable to the entire bucket.

File: docs/objects/object_notifications.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Tigris object notifications allow you to receive notifications via a webhook.
+These events allow you to keep track of when objects are created, updated, or
+deleted for a specific bucket.
+
+## Enable Object Notifications via Console (TODO)
+
+To enable object notifications, you need to set up a webhook in the Tigris
```
Comment (ovaistariq) [inline docs/objects/object_notifications.md:9]: Nice to replace this with a screenshare video similar to how we have in other pages.

File: sidebars.js [hunk 3]
```diff
@@ -74,6 +74,7 @@ const sidebars = {
         "objects/upload-via-html-form",
         "objects/access-objects-via-cookies",
         "objects/acl",
+        "objects/object_notifications",
```
Comment (ovaistariq) [inline sidebars.js:77]: buckets/object_notifications

File: docs/pricing/index.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ Tigris pricing is based on the following components:
 | Data Storage                                  | $0.02/GB/month        |
 | Class A Requests: PUT, COPY, POST, LIST       | $0.005/1000 requests  |
 | Class B Requests: GET, SELECT, and all others | $0.0005/1000 requests |
+| Object Notifications                          | $0.01/1000 events     |
```
Comment (ovaistariq) [inline docs/pricing/index.md:19]: `$0.01/1000 events published`

**Comment by ovaistariq** [comment]
> @ovaistariq updated with pricing and moved to the bucket. Should I add a faq question around what is an event? I'm just waiting to add the loom video for the console.

Yeah, a FAQ question on the pricing page about the event would be good.

**Comment by garrensmith** [comment]
@ovaistariq I've pushed an FAQ section

File: docs/buckets/object_notifications.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+`Authorization: Bearer <token>`
+
+## Notification Types
+
+Tigris currently supports the following notification types:
+
+- `OBJECT_CREATED_PUT`: When an object is created or updated.
+- `OBJECT_DELETED`: When an object is deleted.
+
+:::note Move events will be supported in the future. :::
```
Comment (ovaistariq) [inline docs/buckets/object_notifications.md:52]: The formatting is broken

image

This has to be formatted the following way:

```
:::note

More events will be supported in the future

:::
```

File: docs/objects/object_notifications.md (hunk 2 above)
Comment (tahakhan) [reply docs/objects/object_notifications.md]: https://github.com/tigrisdata/tigris-os-docs/pull/107/commits/047f8190c2d36c8a15ea7bd21018db709f30798b

File: docs/buckets/object_notifications.md (hunk 5 above)
Comment (garrensmith) [reply docs/buckets/object_notifications.md]: Fixed.

**Review by Xe — APPROVED** [review]
Mostly nitpicks/usage semantics that you can punt on

File: docs/buckets/object_notifications.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+| object       | Object | Details of the object involved in the event                  |
+| object.key   | String | The key (path) of the object within the bucket               |
+| object.size  | Number | Size of the object in bytes                                  |
+| object.eTag  | String | Entity tag (ETag) of the object, typically an MD5 hash       |
+
+An example notification payload is:
+
+```json
+{
+  "Records": [
```
[the remark this answers is not in the document]
Comment (garrensmith) [reply docs/buckets/object_notifications.md:78]: Sadly this is correct. This is based off of the AWS events

File: docs/buckets/object_notifications.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+- `OBJECT_CREATED_PUT`: When an object is created or updated.
+- `OBJECT_DELETED`: When an object is deleted.
+
+:::note
+
+More events will be supported in the future.
+
+:::
+
+## Notification Format
```
[the remark this answers is not in the document]
Comment (garrensmith) [reply docs/buckets/object_notifications.md:57]: We don't support that yet. That will be in a future release

File: docs/pricing/index.md [hunk 8]
```diff
@@ -10,12 +10,13 @@ Tigris pricing is based on the following components:
...
-| Data Storage                                  | $0.02/GB/month        |
-| Class A Requests: PUT, COPY, POST, LIST       | $0.005/1000 requests  |
-| Class B Requests: GET, SELECT, and all others | $0.0005/1000 requests |
+| Component                                     | Price                       |
+| --------------------------------------------- | --------------------------- |
+| Data Transfer                                 | Free                        |
+| Data Storage                                  | $0.02/GB/month              |
+| Class A Requests: PUT, COPY, POST, LIST       | $0.005/1000 requests        |
+| Class B Requests: GET, SELECT, and all others | $0.0005/1000 requests       |
+| Object Notifications                          | $0.01/1000 events published |
```
[the remark this answers is not in the document]
Comment (garrensmith) [reply docs/pricing/index.md:19]: I mention that in an FAQ below or is there a better way of saying it here?

File: docs/buckets/object_notifications.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (garrensmith) [reply docs/buckets/object_notifications.md:78]: @himank this has come up a bit. We keep `Records` as uppercase and most others lower camelcase which is the stand. `Records` matches AWS events but it is kind of weird

[the remark this answers is not in the document]
Comment (himank) [reply docs/buckets/object_notifications.md:78]: Let's change it? Initially I was thinking if the top level is same then it is easily replaceable for user but that won't be enough for the end user as for every other storage provider events are custom. Let's then do `records`?

[the remark this answers is not in the document]
Comment (garrensmith) [reply docs/buckets/object_notifications.md:78]: Cool. I agree let's change it to make it more consistent.

File: docs/buckets/object_notifications.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
+# Object Notifications
```
[the remark this answers is not in the document]
Comment (ovaistariq) [reply docs/buckets/object_notifications.md:1]: @garrensmith could you update the filename as above?

File: docs/buckets/object_notifications.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+Tigris object notifications allow you to receive notifications via a webhook.
+These events allow you to keep track of when objects are created, updated, or
+deleted for a specific bucket.
+
+## Enable Object Notifications via Console (TODO)
```
Comment (ovaistariq) [inline docs/buckets/object_notifications.md:7]: Remove `TODO`

File: docs/buckets/object_notifications.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+To enable object notifications, you need to set up a webhook in the Tigris
```
Comment (ovaistariq) [inline docs/buckets/object_notifications.md:9]: Change `Tigris Console` to `Tigris Dashboard`

File: docs/buckets/object_notifications.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+
+<a href="https://www.loom.com/share/37e407ecb5b64cada3aa418e3a2c1df6" target="_blank">
+  <img src="https://cdn.loom.com/sessions/thumbnails/37e407ecb5b64cada3aa418e3a2c1df6-f9556cf7889fc886-full-play.gif" />
+</a>
+
+## Webhook
+
+Object notifications are delivered via a webhook, and obey the following rules:
+
+- Tigris will make an HTTP POST post request to the webhook URL with the event
```
Comment (ovaistariq) [inline docs/buckets/object_notifications.md:22]: `HTTP POST post` -> `HTTP POST`

File: docs/buckets/object_notifications.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+  payload.
+- A `200` status code acknowledges the request was successful.
+- If a `200` status is not received, Tigris will retry for a maximum of 3 times
+  before giving up and marking the notifications as sent.
+- If a webhook request takes longer than 5 seconds the request will be aborted
```
[the remark this answers is not in the document]
Comment (ovaistariq) [reply docs/buckets/object_notifications.md:27]: @garrensmith ^^

File: docs/buckets/object_notifications.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ovaistariq) [reply docs/buckets/object_notifications.md:78]: Have we changed it? And in that case does the example here need to be updated?

**Pull request merged**